pkg/standalone-utils/test/ProtocolFeeSplitter.test.ts

```diff
@@ -0,0 +1,229 @@
+    await vault.grantPermissionsGlobally([withdrawCollectedFeesWithdrawerRole], protocolFeeSplitter);
+
+    const setTreasuryRole = await actionId(protocolFeeSplitter, 'setTreasury');
+    await vault.grantPermissionsGlobally([setTreasuryRole], admin);
+  });
+
+  async function createPool(poolOwnerAddress: string): Promise<Contract> {
+    const receipt = await (
+      await factory.create(
+        NAME,
+        SYMBOL,
+        tokens.addresses,
+        WEIGHTS,
+        assetManagers,
+        POOL_SWAP_FEE_PERCENTAGE,
+        poolOwnerAddress
+      )
+    ).wait();
+
+    const event = expectEvent.inReceipt(receipt, 'PoolCreated');
+    return deployedAt('v2-pool-weighted/WeightedPool', event.args.pool);
+  }
+
+  describe('constructor', () => {
+    it('sets the protocolFeesWithdrawer', async () => {
+      expect(await protocolFeeSplitter.protocolFeesWithdrawer()).to.be.eq(protocolFeesWithdrawer.address);
+    });
+    it('sets the treasury', async () => {
+      expect(await protocolFeeSplitter.treasury()).to.be.eq(treasury.address);
+    });
+  });
+
+  it('changes the treasury', async () => {
+    const receipt = await (await protocolFeeSplitter.connect(admin).setTreasury(newTreasury.address)).wait();
+    expectEvent.inReceipt(receipt, 'TreasuryChanged', { newTreasury: newTreasury.address });
+  });
+
+  describe('default revenue sharing fee', async () => {
+    it('sets default fee', async () => {
+      const newFee = bn(10e16); // 10%
+      const receipt = await (
+        await protocolFeeSplitter.connect(admin).setDefaultRevenueSharingFeePercentage(newFee)
+      ).wait();
+      expectEvent.inReceipt(receipt, 'DefaultRevenueSharingFeePercentageChanged', { revenueSharePercentage: newFee });
+      expect(await protocolFeeSplitter.defaultRevenueSharingFeePercentage()).to.be.eq(newFee);
+    });
+  });
+
+  describe('revenue sharing override', async () => {
+    it('overrides revenue sharing percentage for a pool', async () => {
+      const newFee = bn(50e16); // 50%
+      const receipt = await (
+        await protocolFeeSplitter.connect(admin).setRevenueSharingFeePercentage(poolId, newFee)
+      ).wait();
+      expectEvent.inReceipt(receipt, 'PoolRevenueShareChanged', { poolId: poolId, revenueSharePercentage: newFee });
+    });
+
+    it('reverts with invalid input', async () => {
+      const newFee = bn(100e16); // 100%
+      await expect(
+        protocolFeeSplitter.connect(admin).setRevenueSharingFeePercentage(poolId, newFee)
+      ).to.be.revertedWith('SPLITTER_FEE_PERCENTAGE_TOO_HIGH');
+    });
+  });
+
+  describe('pool beneficiary', async () => {
+    it('reverts if caller is not the pool owner', async () => {
+      await expect(
+        protocolFeeSplitter.connect(liquidityProvider).setPoolBeneficiary(poolId, liquidityProvider.address)
+      ).to.be.revertedWith('SENDER_NOT_ALLOWED');
+    });
```

Comment:
I'd add a test here where the owner successfully sets the beneficiary, testing the return value, event, etc. (in separate tests)